dnf/cli/commands/clean.py

```diff
@@ -102,4 +126,13 @@ def run(self):
 
         patterns = [dnf.repo.CACHE_FILES[t] for t in types]
         count = _clean(cachedir, _filter(files, patterns))
+        counter = 0
+
+        if 'all' in self.opts.type:
+            dirs = list(_dir_tree(cachedir))
+            counter = _clean_dirs(cachedir, _filter(dirs, [
+                r'^%s\.*$' % dnf.repo._CACHEDIR_RE]))
```

Comment:
What is meaning of the r'^%s\.*$' ? I do not understand. It is working? Maybe you want write this r'^%s/.*$'. Or this r'^%s(/.*)?$' if you want remove the repo root directory too.